web/src/components/shared/Ads/Board.tsx

```diff
@@ -28,23 +29,96 @@ export const URL_PARAMS: UrlParams<Omit<AdvertParams, 'lang'>> = {
   isOwnerVerificated: { name: 'verif', set: (v) => `${v}`, get: (v) => Boolean(v) },
 };
 
+export const adTypeToSeo: Record<AdvertType, SeoAdvertType> = {
+  purchase: 'buy',
+  selling: 'sell',
+};
+
+export const seoTypeToAd: Record<SeoAdvertType, AdvertType> = {
+  buy: 'purchase',
+  sell: 'selling',
+};
+
+const isSeoType = (type: string | undefined): type is SeoAdvertType => {
+  return Boolean(type && (type === 'buy' || type === 'sell'));
+};
+
+const getFilterPathParams = (
+  filter?: string,
+): Pick<AdvertParams, 'type' | 'cryptocurrency' | 'currency'> => {
+  const [seoType, cryptocurrency, currency] = filter?.split('-') ?? [];
+
+  return {
+    type: isSeoType(seoType) ? seoTypeToAd[seoType] : DEFAULT_FILTER.type,
```

Comment:
```suggestion
    type: seoTypeToAd[seoType] ?? DEFAULT_FILTER.type,
```
Можно вовсе без проверки `isSeoType`

Reply:
seoType может быть undefined, а ts не даст в индекс вставлять undefined

Reply:
Понял